pages/_document.page.tsx

```diff
@@ -22,6 +22,33 @@ class MyDocument extends Document {
             href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:wght@300;400;700&display=swap"
             rel="stylesheet"
           />
+          {process.env.HELPJUICE_ORIGIN && (
+            <link
+              rel="stylesheet"
+              href={`${process.env.HELPJUICE_ORIGIN}/swifty.css`}
+            />
+          )}
+          {/* The swifty.js script initializes in response to a DOMContentLoaded
+           * event, but that event has fired by the time the script is executed
+           * when using the Next.js <Script> component. To workaround this, we
+           * have to use a native <script> element and cannot use async or
+           * defer. This is not ideal for first-load performance and should be
+           * switched to <Script> as soon as Helpjuice can fix their swifty.js
+           * script.
+           */}
+          {process.env.HELPJUICE_ORIGIN && (
+            <script
+              dangerouslySetInnerHTML={{
+                __html: `window.helpjuice_account_url = '${process.env.HELPJUICE_ORIGIN}';
+window.helpjuice_contact_us_url = '${process.env.HELPJUICE_ORIGIN}/contact-us';
+window.helpjuiceSwiftyConfig = { widgetPosition: 'bottomRight' };
+window.helpjuiceSwiftyUrlMap = {};`,
+              }}
```

Comment:
Good question! I wish that would work, but `_document.page.tsx` only runs on the server, so we don't have access to `window` here. Also we need those variables set before the `swifty.js` script executes. So setting these fields on `window` in a `useEffect` in `_app.page.tsx` for example would be too late because the app mounts after the swifty script executes.

Reply:
Thank you for checking